### DISCUSSION TITLE
Errata, typos, suggestions - "Packages as layers, not groups"

### DISCUSSION THREAD

**Discussion by sleeping-barber** [benbjohnson/wtf#9] [General]
Hi Ben,
I think there is a duplication/context blend at the paragraph **The naive approach**:

> ....
> Second, our HTTP layer can only serve data from SQLite now. What happens if we need to add a caching layer in between? Or how do we support other types of data storage such as Postgres or even storing as JSON on disk? **What if we want to mock our database to test our HTTP layer in isolation?**
> 
> **Finally, we need to run an SQLite database for every HTTP test since there's no abstraction layer to mock it out.** 
> ....

I feel reading this that you wanted to make three points here:

1. the data type names are odd
2. we can't easily change to another data mechanism
3. we can't mock the data layer

Somehow the last sentence is already introducing the third argument.

Cheers!

**Comment by sleeping-barber** [comment]
**Suggestion**

Add example links to the different packagings within the **Repackaging packages** paragraph (e.g. a link to the main application using the SQLite implementation and a test file using the mock.

**Comment by benbjohnson** [comment]
Great suggestions. Thank you @sleeping-barber! I updated the post with the corrections. 🙏
